cmake/host-tools.cmake

```diff
@@ -7,6 +7,19 @@ find_program(
   )
 if(${WEST} STREQUAL WEST-NOTFOUND)
   unset(WEST)
+else()
+  execute_process(
```

Comment:
A comment above execute process here would be very helpful. E.g.

"When west is found check if we are inside a west installation, and if so mark west as not found because X."

Reply:
It kind of sounds like we would want to have two variables, WEST, and IS_INSIDE_WEST_INSTALLATION. But I'm not familiar enough with this domain to be sure.

Reply:
Fixed. I don't think we need the two variables. west is essentially useless when not in an installation. It's like Git outside of a repository.

Reply:
Doesn't the build system use WEST for flashing?

E.g. won't this affect zephyr/cmake/flash/CMakeLists.txt ?

Reply:
> E.g. won't this affect zephyr/cmake/flash/CMakeLists.txt ?

that is already broken, and this PR doesn't change that.
It is actually a very valid question if that should be fixed as well, but that would need a fix in `west`, and might not be so trivial, as `west flash` is an extension command, so it would require at least (as project is mono-repo is managed without `west`):
1) `west` repo cloned somewhere, and path known to it know, e.g. `WEST_DIR`
2) Ability in `west` to load extension commands at different location, as it might not be located together with a `manifest` project

Reply:
> that is already broken

My mistake, I was not aware. Please do fix/remove any broken code.

Reply:
> >    that is already broken

> My mistake, I was not aware. Please do fix/remove any broken code.

Me neither, I only discovered this, because you dragged my attention to this with your comment.
And could see this must have been broken for some time, unfortunately :(

Reply:
actually, flashing should still work if following this guide: https://docs.zephyrproject.org/latest/guides/west/without-west.html